trin-execution/src/execution.rs

```diff
@@ -213,6 +221,61 @@ impl State {
             }
         }
 
+        // check if dao fork, if it is drain accounts and transfer it to beneficiary
+        if block_tuple.header.header.number == get_spec_block_number(SpecId::DAO_FORK) {
+            let mut drained_balance_sum = U256::ZERO;
```

Comment:
This should be in `dao_fork.rs`. I think you will just have to pass `&mut self.database`, which is fine.